pkg/osquery/tables/airport/table_darwin.go

```diff
@@ -0,0 +1,262 @@
+	   i got spaces! a0:a0:a0:a0:a0:a0 -92  108     Y  US WPA(PSK/AES,TKIP/TKIP) RSN(PSK/AES,TKIP/TKIP)
+	       no-spaces b1:b1:b1:b1:b1:b1 -91  116     N  EU RSN(PSK/AES/AES)
+	*/
+
+	scanner := bufio.NewScanner(reader)
+	scanner.Split(bufio.ScanLines)
+
+	// use this to store the cells of the header row inclucing whitespaces
+	// we can use the length of these to determine the column widths
+	// we need this so we can handle whitespaces in SSID and Security column
+	var rawHeaders []string
+	var results []map[string]interface{}
+
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		if len(line) == 0 || strings.Trim(line, " ") == "" {
+			continue
+		}
+
+		// first popoulate the headers including spaces
+		// this also determines the column widths
+		if rawHeaders == nil {
+			rawHeaders = rawSpaceSeparatedWords(line)
+
+			if len(rawHeaders) < 2 {
+				continue
+			}
+
+			// the last header: "SECURITY (auth/unicast/group)" has a space in it, so combine them
+			rawHeaders[len(rawHeaders)-2] = fmt.Sprintf("%s %s", rawHeaders[len(rawHeaders)-2], rawHeaders[len(rawHeaders)-1])
+			rawHeaders = rawHeaders[:len(rawHeaders)-1]
+			continue
+		}
+
+		rowData := make(map[string]interface{})
+
+		startRuneIndex := 0
+		for headerIndex, header := range rawHeaders {
+
+			key := strings.Trim(header, " ")
+
+			endRuneIndex := startRuneIndex + len(header)
+
+			// if last header or runes left are less than what column should be, use remainder of line
+			// for example: WPA(PSK/AES,TKIP/TKIP) RSN(PSK/AES,TKIP/TKIP)
+			if headerIndex == len(rawHeaders)-1 || len(line) < endRuneIndex {
+				rowData[key] = strings.Trim(line[startRuneIndex:], " ")
+				break
+			}
+
+			value := strings.Trim(line[startRuneIndex:endRuneIndex], " ")
+			// trim whitespaces from header value
+			rowData[key] = value
+			startRuneIndex = endRuneIndex + 1
+		}
+
+		results = append(results, rowData)
+	}
+
+	return results
+}
+
+func rawSpaceSeparatedWords(line string) []string {
+	separatorIndexes := columnSeparatorIndexes(line)
+
+	var words []string
+	lastSeparatorIndex := 0
+
+	for _, separatorIndex := range separatorIndexes {
+		words = append(words, line[lastSeparatorIndex:separatorIndex])
+		lastSeparatorIndex = separatorIndex + 1
+	}
+
+	words = append(words, line[lastSeparatorIndex:])
+
+	return words
+}
+
+func columnSeparatorIndexes(line string) []int {
+	var indexes []int
+
+	lastSeperatorIndex := 0
+
+	for {
+		trimmedLeft := strings.TrimLeft(line, " ")
+
+		endIndex := indexOfLastSpaceBeforeNonSpace(trimmedLeft)
+		if endIndex == -1 {
+			return indexes
+		}
+
+		leadingSpacesCount := len(line) - len(trimmedLeft)
+
+		separatorIndex := lastSeperatorIndex + leadingSpacesCount + endIndex
+		indexes = append(indexes, separatorIndex)
+
+		lastSeperatorIndex = separatorIndex
+
+		line = trimmedLeft[endIndex:]
+	}
+}
+
+func indexOfLastSpaceBeforeNonSpace(str string) int {
+	for i := 0; i < len(str)-1; i++ {
```

Comment:
Lets say I have:

`a b`

that would return `0` right?

but we want it to return `1` since that would be the index of the last space before a non space ... maybe I'm missing something